### ISSUE TITLE
Feature request: add safeParse option

### ISSUE THREAD

**Issue by am29d** [aws-powertools/powertools-lambda-typescript#2204]
### Use case

If you don't want Zod to throw errors when validation fails, use `.safeParse`. This method returns an object containing either the successfully parsed data or a ZodError instance containing detailed information about the validation problems.

There are cases when customers used `safeParse` in their previous implementation and expect to have this option in the powertools parser utility. Thus, we should add an option to that will determine wether we call `parse` or `safeParse` internally in built-in schemas and `envelopes`

### Solution/User Experience

We can expand the `ParserOptions` by adding `safeParse` argument, and call it directly on decorator

```typescript
import { SqsSchema } from '@aws-lambda-powertools/parser/schemas';
import { SqsEvent } from '@aws-lambda-powertools/parser/types';

export type ParserOptions<S extends ZodSchema> = {
  schema: S;
  envelope?: Envelope;
  safeParse?: boolean;
};


class Lambda implements LambdaInterface {
  @parser({ schema: SqsSchema, safeParse: true })
  public async handler(
    event: SqsEvent,
    _context: Context
  ): Promise<unknown> {
    return event;
  }

```

This would also require an expansion of the envelopes to support `safeParse` option.

### Acknowledgment

- [X] This feature request meets [Powertools for AWS Lambda (TypeScript) Tenets](https://docs.powertools.aws.dev/lambda/typescript/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Python](https://github.com/aws-powertools/powertools-lambda-python/), [Java](https://github.com/aws-powertools/powertools-lambda-java/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### Future readers

Please react with 👍 and your use case to help us understand customer demand.

**Comment by dreamorosi** [comment]
Will this value be enabled by default?

**Comment by dreamorosi** [comment]
This is now released under [v2.1.0](https://github.com/aws-powertools/powertools-lambda-typescript/releases/tag/v2.1.0) version!
